scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala

```diff
@@ -233,8 +233,10 @@ class LegacyScanner(input: Input, dialect: Dialect) {
     }
 
     @inline
-    def reportIllegalCharacter(): Unit = curr
-      .setInvalidToken(s"illegal character '\\u${"%04x".format(ch)}'")
+    def reportIllegalCharacter(): Unit = {
+      val output = "\\u" + Integer.toHexString(ch | 0x10000).substring(1);
```

Comment:
This is a bit less clear but doesn't involve regex underneath

Reply:
can you, perhaps, add a comment in the code explaining what this does?

Reply:
Maybe? The stack trace goes later like:
```
at java.lang.String.format(java.base@11.0.24/String.java:2897)
	at scala.collection.StringOps$.format$extension(StringOps.scala:991)
	at scala.meta.internal.tokenizers.LegacyScanner.reportIllegalCharacter$1(LegacyScanner.scala:237)
```

It's here https://gist.github.com/felher/9939f409ab3f7d0df5e5abe8f06ec1e2

Reply:
Ok, I changed it to not use `s"` and added proper explanations.